### PR TITLE
Ship Autoload ui fixes 2

### PR DESCRIPTION
**Pull request by mkerensky**
Improvements to the UI of assigning units to ships from the TOE tab
Fixes some bugs that causes the Assign to Ship menu item to display when it shouldn't, as well as not display when it should.

### CODE REVIEW

File: MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java [hunk 1]
```diff
@@ -1168,15 +1172,30 @@ private void maybeShowPopup(MouseEvent e) {
                 JMenu i_trn = new JMenu(TOEMouseAdapter.INFANTRY_CARRIERS);
                 JMenu a_trn = new JMenu(TOEMouseAdapter.ASF_CARRIERS);
                 JMenu sc_trn = new JMenu(TOEMouseAdapter.SC_CARRIERS);
+                JMenu singleUnitMenu = new JMenu("");
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java]: Why are you initializing this with a "" instead of just not including text?

Comment (mkerensky) [reply MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java]: No reason... <whistles>    Fixed.

File: MekHQ/src/mekhq/gui/utilities/StaticChecks.java [hunk 2]
```diff
@@ -405,6 +405,42 @@ public static boolean doAllUnitsHaveC3Master(Vector<Unit> units) {
...
+    
+    /**
+     * Used to test a selection of Units provided by the player and determine whether they all share a designated unitType. 
+     * @param units Vector of units that the player has selected
+     * @return false if any unit in the passed-in Vector does not have the specified unit type
+     */
+    public static boolean areAllUnitsSameType(Vector<Unit> units, int unitType) {
+        // For our purposes, a selection of tanks isn't the same unless they're all the same weight class
+        // Set this based on the first unit in the selection
+        double firstUnitWeight = units.get(0).getEntity() != null ? units.get(0).getEntity().getWeight() : 0;
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: This should exit early if `units` is empty.

**Comment by NickAragua** [comment]
Sorry this was left hanging for so long, could you do the pending merge? Then we'll get it merged in.

File: MekHQ/src/mekhq/gui/utilities/StaticChecks.java [hunk 3]
```diff
@@ -405,6 +405,42 @@ public static boolean doAllUnitsHaveC3Master(Vector<Unit> units) {
...
+        boolean light = firstUnitWeight > 0 && firstUnitWeight <= 50;
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: Is the > 0 necessary? Won't they all have at least some weight (if you have some unit).

File: MekHQ/src/mekhq/gui/utilities/StaticChecks.java [hunk 4]
```diff
@@ -405,6 +405,42 @@ public static boolean doAllUnitsHaveC3Master(Vector<Unit> units) {
...
+        boolean heavy = firstUnitWeight > 50 && firstUnitWeight <= 100;
+        boolean superheavy = firstUnitWeight > 100 && firstUnitWeight <= 200;
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: Is there something > 200 for units?

Comment (mkerensky) [reply MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: There is not. I know you can construct such units, but the SH Vehicle *bay* only holds up to 200 tons - anything heavier is Cargo - so I think I'm ok here.

Comment (Windchild292) [reply MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: There are a few examples, like the Koi and Ryu WiGE.

Comment (neoancient) [reply MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: But mkerensky is correct that vehicles over 200 tons cannot be transported in a bay.

File: MekHQ/src/mekhq/gui/utilities/StaticChecks.java (hunk 2 above)
Comment (mkerensky) [reply MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: Fixed.  I thought I only called this in the first place after checking that in TOEMouseAdapter, but that is only done in one of two locations.

File: MekHQ/src/mekhq/gui/utilities/StaticChecks.java (hunk 3 above)
Comment (mkerensky) [reply MekHQ/src/mekhq/gui/utilities/StaticChecks.java]: Seemed like a good idea, in case we managed to get a botched unit with negative or 0 weight

**Comment by mkerensky** [comment]
I think I've got everything fixed here. I also found a couple of cases where I was checking for a null ArrayList instead of an empty one and fixed those.

File: MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java [hunk 5]
```diff
@@ -280,22 +281,27 @@ public void actionPerformed(ActionEvent action) {
...
             }
         }
         if (command.contains(UNASSIGN_FROM_SHIP)) {
-            for (Unit u : units) {
-                for (UUID oldShipId : u.getTransportShipId().keySet()) {
-                    Unit oldShip = gui.getCampaign().getUnit(oldShipId);
-                    if (oldShip != null) {
-                        oldShip.unloadFromTransportShip(u);
-                        MekHQ.triggerEvent(new UnitChangedEvent(oldShip));
+            if (!units.isEmpty()) {
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java]: This is superfluous with the for loop below. It'll do nothing if units is empty.

**Pull request merged**